apricot/local/servicehttp.go

```diff
@@ -0,0 +1,109 @@
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * In applying this license CERN does not waive the privileges and
+ * immunities granted to it by virtue of its status as an
+ * Intergovernmental Organization or submit itself to any jurisdiction.
+ */
+
+package local
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/AliceO2Group/Control/configuration"
+	"github.com/gorilla/mux"
+)
+
+type HttpService struct {
+	svc configuration.Service
+}
+
+func (httpsvc *HttpService) ApiGetFlps(w http.ResponseWriter, r *http.Request) {
+	keyPrefix := "o2/hardware/flps/"
+	httpsvc.ApiGetClusterInformation(w, r, keyPrefix)
+}
+
+func (httpsvc *HttpService) ApiGetDetectorFlps(w http.ResponseWriter, r *http.Request) {
+	keyPrefix := "o2/hardware/detectors/TST/flps/"
+	httpsvc.ApiGetClusterInformation(w, r, keyPrefix)
+}
+
+func (httpsvc *HttpService) ApiGetClusterInformation(w http.ResponseWriter, r *http.Request, keyPrefix string) {
+	queryParam := mux.Vars(r)
+	format := ""
+	format = queryParam["format"]
+	if format == "" {
+		format = "text"
+	}
+	keys, err := httpsvc.svc.GetHostInventory(keyPrefix)
+	if err != nil {
+		log.WithError(err).Fatal("Error, could not retrieve host list.")
+	}
+	switch format {
+	case "json":
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusOK)
+		hosts, err := json.MarshalIndent(keys, "", "\t")
+		if err != nil {
+			log.WithError(err).Fatal("Error, could not marshal hosts.")
+		}
+		fmt.Fprintln(w, string(hosts))
+	case "text":
+		w.Header().Set("Content-Type", "text/plain")
+		w.WriteHeader(http.StatusOK)
+		for _, hostname := range keys {
+			fmt.Fprintf(w,"%s\n", hostname)
+		}
+	default:
+		w.Header().Set("Content-Type", "text/plain")
+		w.WriteHeader(http.StatusOK)
+		for _, hostname := range keys {
+			fmt.Fprintf(w,"%s\n", hostname)
+		}
+	}
+}
+
+func NewHttpService(service configuration.Service) (svr *http.Server) {
+	router := mux.NewRouter()
+	httpsvc := &HttpService{
+		svc: service,
+	}
+	httpsvr := &http.Server{
+		Handler:      router,
+		Addr:         ":47188",
+		WriteTimeout: 15 * time.Second,
+		ReadTimeout:  15 * time.Second,
+	}
+	go func() {
+		apiFlps := router.PathPrefix("/inventory/flps").Subrouter()
+		apiFlps.HandleFunc("", httpsvc.ApiGetFlps).Methods(http.MethodGet)
+		apiFlps.HandleFunc("/", httpsvc.ApiGetFlps).Methods(http.MethodGet)
+		apiFlps.HandleFunc("/{format}", httpsvc.ApiGetFlps).Methods(http.MethodGet)
+		apiDetectorFlps := router.PathPrefix("/inventory/detectors/TST/flps").Subrouter()
```

Comment:
`TST` here should be a variable. The allowed values are in `common/systemid/systemid.go`, and they are an enum, so no strings for now. You should not use the enums, this is just to give you an idea of what could be found as input instead of `TST`.

For example the route handler for `/inventory/detectors/ITS/flps` should return whatever's to be found in `o2/hardware/detectors/ITS/flps/`.